### ISSUE TITLE
Write only mode for AWS

### ISSUE THREAD

**Issue by barryvdh** [spatie/laravel-backup#188]
Is it possible to use a 'write only' mode, without listing/reading, for Amazon AWS?

Eg. I have a policy that only allows a 'PUT' command, so that a backup service can only ever put new backups and not read/delete them.

This combined with the Amazon versioning + lifecycle to move to glacier, makes sure that files are never deleted (unless the period is over). Do you have a recommended way to do this?

**Comment by freekmurze** [comment]
That's an interesting strategy.  

I'm thinking that you should be able to take a backup as the package will only do [a write operation](https://github.com/spatie/laravel-backup/blob/master/src/BackupDestination/BackupDestination.php#L88-L92).

All other commands (cleanup, listing, monitoring, ...) will probably not work as there is some reading involved.

Test it out, and let me know how it goes.

**Comment by barryvdh** [comment]
I got a problem earlier when I tried, should have mentioned that. It tried again and looked further.

Before creating a backup, `isReachable()` is called, which lists all files. This is not allowed (in my current policy).
Is that for performance, or a different reason? Looking in the code, it seems that first the Zip is created (the heavy part), and then the destinations are checked and immediately after the stream is written. If a destination is not reachable, the stream operation would fail directly anyway, right? So is the directory listing even necessary?

Second thing is that the filename is not configurable, right? That would be needed to use AWS versioning (same filename). This could be fixed by allowing an optional filename flag perhaps?

```
php artisan backup:run --only-db --filename=db.zip 
php artisan backup:run --only-files --filename=media.zip 
```

If you want I can create a PR for both issues.
